appscale/tools/remote_helper.py

```diff
@@ -850,6 +886,28 @@ def wait_for_machines_to_finish_loading(cls, host, keyname):
         time.sleep(cls.WAIT_TIME)
 
 
+  @classmethod
+  def terminate_spawned_instances(cls, spawned_instance_ids, agent, params):
+    """ Shuts down instances specified. For use when AppScale has failed to
+    start all the instances for the deployment since we do not check or clean
+    any local files.
+
+    Args:
+      spawned_instance_ids: A list of instance ids we have started that
+        should be terminated.
+      agent: The agent to call terminate instance with.
+      params: Agent parameters.
+    """
+    terminate_params = params.copy()
+    terminate_params[agent.PARAM_INSTANCE_IDS] = spawned_instance_ids
+    for _ in range(len(spawned_instance_ids)):
```

Comment:
Why is this running the terminate call multiple times (based on the number of started machines)?

Reply:
You are right, it does not need a loop because the terminate instances method uses a list of instances to delete anyways.